Mixpanel/Mixpanel.m

```diff
@@ -662,6 +662,13 @@ - (void)flushQueue:(NSMutableArray *)queue endpoint:(NSString *)endpoint
         NSString *response = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
         if ([response intValue] == 0) {
             MixpanelError(@"%@ %@ api rejected some items", self, endpoint);
+        } else if ([response intValue] == 1) {
+            // Success
+        } else {
+            // Another failure occurred. We expected a response of either `0` or `1`, but received something else.
+            // Maybe we were on hotel WiFi and had the request intercepted. Don't drop the data but retry again later.
+            // XXX Ideally we'd do something more intelligent, maybe by looking at `urlResponse` more closely.
+            break;
```

Comment:
I am not sure exactly how you want to "change the loop". The loop is `while ([queue count] > 0)`, and you only have `response` later in its body (the request is performed synchronously). Also note that you *don't* want to break on `!= 1`, you want to break if `response.intValue != 0 && response.intValue != 1`.

Care to elaborate? :-)